src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,140 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    static class Node<T> {
+        private T element;
+        private Node<T> next;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (size == 0) {
+            first = last = newNode;
+        } else {
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddByIndex(index);
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            last.next = newNode;
+            last = newNode;
+        } else {
+            Node<T> currentNode = getNodeByIndex(index - 1);
+            newNode.next = currentNode.next;
+            currentNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = getNodeByIndex(index);
+        T element = currentNode.element;
+        currentNode.element = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T element;
+        if (index == 0) {
+            element = first.element;
+            first = first.next;
+            if (first == null) {
+                last = null;
+            }
+        } else {
+            Node<T> currentNode = getNodeByIndex(index - 1);
+            element = currentNode.next.element;
+            currentNode.next = currentNode.next.next;
+            if (index == size - 1) {
+                last = currentNode;
+            }
+        }
+        size--;
+        return element;
     }
 
     @Override
     public boolean remove(T object) {
+        int index = 0;
+        for (Node<T> currentNode = first; currentNode != null; currentNode = currentNode.next) {
+            if (equalsElemets(currentNode.element, object)) {
+                remove(index);
```

Comment:
let's call `unlink(Node<T> node)` here instead `remove(index);`, see the previous comment. 
In current implementation the loop for searching element is called twice (here and later in `getNodeByIndex()`)  , that reduce the performance.